src/gluonnlp/model/transformer.py

```diff
@@ -360,6 +361,13 @@ def __init__(self, attention_cell='multi_head', num_layers=2,
         self._scaled = scaled
         self._use_layer_norm_before_dropout = use_layer_norm_before_dropout
         self._scale_embed = scale_embed
+        self._support_arange_like = False
+        try:
+            self._support_arange_like = bool(ndarray.contrib.arange_like)
+        except AttributeError:
```

Comment:
Other parts of the library will require mxnet 1.5 too. So you don't need provide a workaround for <1.5 IMO

Reply:
I agree. We need to force the user to upgrade the version of MXNet.

Reply:
this op is post 1.5 if i remember correctly

Reply:
Yes, it's not in 1.5.0 release.

Reply:
Can you use F.arange(..., infer_range=True) instead? See https://github.com/dmlc/gluon-nlp/pull/846/files#diff-deaa65ee7b0ea2b813a77275105de177R199-R200 for an example

Reply:
Thank you @leezu. Does the below method look good to you? Since it will be sightly slower than the direct arange_like way, I will put it into the case when arange_like is not supported to replace the max_length way. @eric-haibin-lin 

```
import mxnet as mx
import numpy as np

a = mx.sym.Variable('a')
b = mx.sym.slice(a, begin=(0, 0, 0), end=(1, None, 1))
c = mx.sym.reshape(b, shape=(-1,))
d = mx.sym.zeros_like(c) + mx.sym.arange(start=0, stop=None, infer_range=True)

mod = d.simple_bind(mx.cpu(), a=(3,4,5), grad_req='null')
mod.arg_arrays[0][:] = np.random.normal(size=mod.arg_arrays[0].shape)
out = mod.forward(is_train=False)
print(out[0])
```

Reply:
Yes, using `infer_range=True` to avoid the `max_length` requirement looks good.
Why do you mean it's slightly slower? Due to `zeros_like(c)`?

Reply:
"arange_like" vs. "slice" + "zeros_like" + "elemwise_add". Here is a quick benchmark on my sever:

```
# shape = (32, 128, 1024)
infer_range true: 0.207459 ms
arange_like: 0.096955 ms
```